pandas/io/formats/printing.py

```diff
@@ -503,7 +505,7 @@ def _justify(
 
 
 def format_object_attrs(
-    obj: Sequence, include_dtype: bool = True
+    obj: Union[Sequence, AnyArrayLike], include_dtype: bool = True
```

Comment:
AnyArrayLike doesnt satisfy Sequence?

Reply:
not at the moment, xref #28770

Reply:
It looks like the issue here actually is that the docstring states iterable, but len is used in the function code and is the only requirement.

```
>>> from pandas.io.formats.printing import format_object_attrs
>>>
>>> format_object_attrs(pd.array([1, 2, 3] * 100, dtype="Int64"))
[('dtype', "'Int64'"), ('length', 300)]
>>>
>>> format_object_attrs(pd.Index([1, 2, 3] * 100))
[('dtype', "'int64'"), ('length', 300)]
>>>
>>> format_object_attrs("foo" * 100)
[('length', 300)]
>>>
>>> class MyClass:
...     def __len__(self):
...         return 300
...
>>> format_object_attrs(MyClass())
[('length', 300)]
>>>
```

I'll type as Sized for now to be as permissive as possible to faciliate re-usability, but AFAICT, format_object_attrs is only ever called with an Index.

(so alternatively could just use Index as type for obj, and maybe reduce the hasattr usages or replace with isinstance(obj, MultiIndex))